packages/framework/src/Component/Elasticsearch/ElasticsearchStructureManager.php

```diff
@@ -87,7 +87,7 @@ public function getVersionedIndexName(int $domainId, string $index): string
     public function getCurrentIndexName(string $aliasName): string
     {
         $indexes = $this->client->indices();
-        if ($indexes->exists(['index' => $aliasName])) {
+        if (!$indexes->existsAlias(['name' => $aliasName])) {
```

Comment:
I would probably add ` && $indexes->exists(['index' => $aliasName]` to the condition, otherwise `ElasticsearchStructureManager::getCurrentIndexName('blabla')` will just return `'blabla'`